source/tyr/tyr/resources.py

```diff
@@ -943,3 +955,108 @@ def delete(self, billing_plan_id=None):
             logging.exception("fail")
             raise
         return ({}, 204)
+
+class Autocomplete(flask_restful.Resource):
+    def get(self, shape=None):
+        if shape:
+            autocomplete = models.Autocomplete.query.filter_by(shape=shape).first_or_404()
+            return marshal(autocomplete, autocomplete_fields)
+        else:
+            autocompletes = models.Autocomplete.query.all()
+            return marshal  (autocompletes, autocomplete_fields)
+
+    def post(self):
+        parser = reqparse.RequestParser()
+        parser.add_argument('shape', type=unicode, required=True,
+                            case_sensitive=False, help='shape is required',
+                            location=('json', 'values'))
+        parser.add_argument('street', type=str, required=False, default='BANO',
+                            help='source for street: [BANO, OSM]',
+                            location=('json', 'values'),
+                            choices=['BANO', 'OSM'])
```

Comment:
You should share the enum between the models and here, else one day we will forgot to update one of them